src/Sulu/Bundle/AdminBundle/Resources/js/views/FormOverlayList/tests/FormOverlayList.test.js

```diff
@@ -0,0 +1,276 @@
+import ResourceFormStore from '../../../containers/Form/stores/ResourceFormStore';
+import Form from '../../../containers/Form';
+import ErrorSnackbar from '../ErrorSnackbar';
+
+const React = mockReact;
+
+jest.mock('../../List', () => class ListMock extends mockReact.Component {
+    listStore = {};
+
+    render() {
+        return <div>list view mock</div>;
+    }
+});
+
+jest.mock('../../../containers/Form', () => class ListMock extends mockReact.Component {
+    render() {
+        return <div>form container mock</div>;
+    }
+});
+
+jest.mock('../../../utils/Translator', () => ({
+    translate: jest.fn((key) => key),
+}));
+
+jest.mock('../../../stores/ResourceStore', () => jest.fn(
+    (resourceKey, itemId) => {
+        return {
+            id: itemId,
+        };
+    }
+));
+jest.mock('../../../containers/Form/stores/ResourceFormStore', () => jest.fn(
+    (resourceStore) => {
+        return {
+            id: resourceStore.id,
+        };
+    }
+));
+
+let router;
+beforeEach(() => {
+    router = {
+        attributes: {
+            id: 'test-id',
+            category: 'category-id',
+        },
+        route: {
+            options: {
+                adapters: ['table'],
+                addRoute: 'addRoute',
+                listKey: 'test-list-key',
+                formKey: 'test-form-key',
+                addOverlayTitle: 'app.add_overlay_title',
+                editOverlayTitle: 'app.edit_overlay_title',
+                resourceKey: 'test-resource-key',
+                toolbarActions: ['sulu_admin.add'],
+                routerAttributesToListStore: {0: 'category', 'parentId': 'id'},
+                routerAttributesToFormStore: {0: 'category', 'parentId': 'id'},
+            },
+        },
+    };
+});
+
+test('View should render with closed overlay', () => {
+    const formOverlayList = mount(<FormOverlayList router={router} />);
+
+    expect(formOverlayList.render()).toMatchSnapshot();
+});
+
+test('View should render with opened overlay', () => {
+    const formOverlayList = mount(<FormOverlayList router={router} />);
+    formOverlayList.instance().createFormOverlay(undefined, 'test-list-key');
+    formOverlayList.update();
+
+    expect(formOverlayList.render()).toMatchSnapshot();
+    expect(formOverlayList.find(Overlay).render()).toMatchSnapshot();
+});
+
+test('Should pass correct props to List view', () => {
+    const formOverlayList = mount(<FormOverlayList router={router} />);
+    const list = formOverlayList.find(List);
+
+    expect(list.props()).toEqual(expect.objectContaining(formOverlayList.props()));
+    expect(list.props().onItemAdd).toBeDefined();
+    expect(list.props().onItemClick).toBeDefined();
+});
+
+test('Should construct ResourceStore and ResourceFormStore with correct parameters on item-add callback', () => {
+    const formOverlayList = mount(<FormOverlayList router={router} />);
+    formOverlayList.find(List).props().onItemAdd();
```

Comment:
Why not using some click as well? Can you check the other tests as well? This way the `onItemAdd` prop might not be correctly wired, and nothing works at all :confused:

Reply:
As explained above, there is no clickable element mounted because the `List` view is mocked.
But in general, applying a click event to the `List` view would basically test the internals of the `List` component, wouldnt it?

In my opinion, testing if given callbacks are fired when respective elements are clicked should happen in the test-suite of the respective components. Otherwise we will implicitly test these implementations in many places and many tests will fail when the wiring inside an unrelated component breaks.